src/APICore.ts

```diff
@@ -69,14 +69,17 @@ export class APICore {
   }
 
   private get requestHeaders(): Record<string, string> {
+    const {version} = require('../package.json');
+
     const authorizationHeader = {
       Authorization: `Bearer ${this.accessToken}`,
     };
 
     const documentsRequestHeaders = {
       ...authorizationHeader,
-      'Content-Type': 'application/json',
       Accept: 'application/json',
+      'Content-Type': 'application/json',
+      'User-Agent': `CoveoNodeJSSDK/${version}`,
```

Comment:
nit: I guess the name comes from the google doc and is not ours to decide - but think going from generic to specific would be better - e.g. `CoveoSDKNodeJS` - also moves the common parts of the name first to form a common prefix.